File: abc/abc167/abc167_b.fs
// https://atcoder.jp/contests/abc167/submissions/13103963

let [| A; B; C; K |] = stdin.ReadLine().Split() |> Array.map int

if A >= K then K
elif A + B >= K then A
else A - (K - A - B)
|> stdout.WriteLine

File: abc/abc167/abc167_b.cs
// https://atcoder.jp/contests/abc167/submissions/13103957

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var A = input[0];
        var B = input[1];
        var C = input[2];
        var K = input[3];

        if (A >= K)
        {
            Console.WriteLine(K);
        }
        else if (A + B >= K)
        {
            Console.WriteLine(A);
        }
        else
        {
            Console.WriteLine(A - (K - A - B));
        }
    }
}
